packages/yew/src/functional/hooks/use_ref.rs

```diff
@@ -124,8 +123,13 @@ where
 ///         </div>
 ///     }
 /// }
-///
 /// ```
+///
+/// # Tip
+///
+/// When conditionally rendering elements you can use `NodeRef` in conjunction with `use_effect_with_deps`
+/// to perform actions each time an element is rendered and just before its going to be removed from the
+/// DOM.
```

Comment:
> and just before its going to be removed from the DOM.

Is the "just before" part true? I always assumed the `NodeRef` gets set to `None` *after* the element already got removed from the DOM. Not sure where this part of the doc is coming from.

Reply:
Can you test it?

I think it's possible to just put the node ref usage in the destructor of use_effect_with_deps and use it

Reply:
Ah! in that case, I misread the sentence as meaning "just before the referenced element is removed", but it means "just before the component the `use_ref` is used in is removed".